src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -39,4 +51,108 @@ public int size() {
     public List<Section> get() {
         return this.sections;
     }
+
+    public boolean containsStation(Station station) {
+        return sortedLineStations().contains(station);
+    }
+
+    public int indexOfStation(Station station) {
+        return sortedLineStations().indexOf(station);
+    }
+
+    public List<StationResponse> allStationResponses() {
+        return sortedLineStations().toStationResponses();
+    }
+
+    private void validateContainsUpDownStation(Section section) {
+        if (this.sections.isEmpty()) return;
+        if (containsStation(section.getUpStation()) && containsStation(section.getDownStation())) {
+            throw new IllegalArgumentException(ALREADY_EXIST_UP_DOWN_STATION);
+        }
+    }
+
+    private LineStations sortedLineStations() {
+        List<Station> sortedAllStations = new ArrayList<>();
+        sortedAllStations.add(findEdgeUpSection().getUpStation());
+        for (int i = 0; i < this.sections.size(); i++) {
+            Station nextStation = findSectionByUpStation(sortedAllStations.get(i))
+                    .map(Section::getDownStation)
+                    .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_SORTED_NEXT_STATION));
+            sortedAllStations.add(nextStation);
+        }
+        return LineStations.from(sortedAllStations);
+    }
+
+    private boolean isEdgeSection(Section addedSection) {
+        return isEdgeUpSection(addedSection) || isEdgeDownSection(addedSection);
+    }
+
+    private boolean isEdgeUpSection(Section addedSection) {
+        return findEdgeUpSection().getUpStation().equals(addedSection.getDownStation());
+    }
+
+    private boolean isEdgeDownSection(Section addedSection) {
+        return findEdgeDownSection().getDownStation().equals(addedSection.getUpStation());
+    }
+
+    private Section findEdgeUpSection() {
+        List<Station> downStations = this.sections.stream()
+                .map(Section::getDownStation)
+                .collect(Collectors.toList());
+        return this.sections.stream()
+                .filter(section -> !section.isIncludedUpStation(downStations))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_EDGE_UP_SECTION));
+    }
+
+    private Section findEdgeDownSection() {
+        List<Station> upStations = this.sections.stream()
+                .map(Section::getUpStation)
+                .collect(Collectors.toList());
+        return this.sections.stream()
+                .filter(section -> !section.isIncludedDownStation(upStations))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_EDGE_DOWN_SECTION));
+    }
+
+    private void relocateConnectSection(Section addedSection) {
+        if (!this.sections.isEmpty() && !isEdgeSection(addedSection)) {
```

Comment:
`!this.sections.isEmpty() && !isEdgeSection(addedSection)` 로직을 메서드로 추출하면 좋을 것 같아요.
```suggestion
        if (addable(addedSection)) {
```